### PR TITLE
refactor(aws):  Refactoring to better encapsulate Asg config related fields and separate launch template/launch config code

### PR DESCRIPTION
**Pull request by pdk27**
## Summary
  - refactored config related to ASG creation into ASGWorker.AsgConfiguration (previously, class fields of AsgWorker)
  - removed usage of LaunchConfigurationSettings in LaunchTemplate code path, replaced it with AsgConfiguration
     - facilitates for easier addition of LT related features without an expanding list of method parameters
  - moved security group related functions to SecurityGroupService (previously in DefaultLaunchConfigurationBuilder)
  - extracted common config setup code to a helper, out of DefaultLaunchConfigurationBuilder
  - added/updated unit tests

## Related Issue
https://github.com/spinnaker/spinnaker/issues/5989

## Background / Upcoming
As part of adding AWS EC2 support for multiple instance types and instance diversification, some refactoring is necessary: 
* Current PR - ASG configuration can now be passed around and is not just confined to _AutoScalingWorker_
* [Next PR (WIP)](https://github.com/pdk27/clouddriver/commit/b96aa49782820988d6df948aafccd573e7280572) - ASG builders will be added to separate ASG request building and ASG creation from conditional logic and dynamic configuration logic in _AutoScalingWorker_
* Next to next PR (WIP) - Support for ASG creation with mixed instances policy (currently, launch template and launch config options are supported) - https://docs.aws.amazon.com/autoscaling/ec2/APIReference/API_MixedInstancesPolicy.html

## Tests 
Manual local tests done to verify creation of ASG with various input parameters

### CODE REVIEW

**Comment by pdk27** [comment]
@spinnaker/reviewers @cfieber

File: clouddriver-aws/src/test/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/AsgConfigHelperSpec.groovy [hunk 1]
```diff
@@ -0,0 +1,207 @@
+/*
```
Comment (robzienert) [inline clouddriver-aws/src/test/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/AsgConfigHelperSpec.groovy:1]: Please do not introduce tabs into the codebase.

**Review by robzienert — APPROVED** [review]
A couple small things, otherwise looks good.

File: clouddriver-aws/src/test/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/AsgConfigHelperSpec.groovy (hunk 1 above)
Comment (cfieber) [reply clouddriver-aws/src/test/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/AsgConfigHelperSpec.groovy:1]: I'd have expected spotless to fix that, I wonder if it's not looking at the tests (or does it not do groovy at all?)

**Comment by cfieber** [comment]
Thanks for the refactors!

**Review by cfieber — APPROVED** [review]

File: clouddriver-aws/src/test/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/AsgConfigHelperSpec.groovy (hunk 1 above)
Comment (pdk27) [reply clouddriver-aws/src/test/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/AsgConfigHelperSpec.groovy:1]: @cfieber You are right! `spotlessCheck` did not identify this file as a violator.

**Pull request merged**